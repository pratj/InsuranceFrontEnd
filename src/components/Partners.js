import { Grid, Typography } from '@material-ui/core'
import React, {useState, useEffect} from 'react'
import { Carousel } from 'react-responsive-carousel'

function Partners() {

    const [partners, setPartners] = useState([])

    useEffect(() => {
        setPartners([
            {
                "image": "https://cms-img.coverfox.com/tata%20aig.jpg",
                "partner": "Tata AIG"
            },
            {
                "image": "https://www.bajajallianz.com/content/dam/bagic/bagic-lazyloading/lazyLoadingBajajLogo.jpg",
                "partner": "Bajaj Allianz"
            },
            {
                "image": "https://static.businessworld.in/article/article_extra_large_image/1575458276_o93UrT_HDFC_ERGO_launches_Inherent_Defects_Insurance_Policy.jpg",
                "partner": "HDFC ERGO"
            }
        ])
    }, [])

    return (
        <Grid container>
            <Grid item xs={false} sm={2}></Grid>
            <Grid item xs={12} sm={8}>
                <Typography variant="h3" style={{ textAlign: 'center', color:"#b58500"}}><b>Our Partners</b></Typography>
                <Carousel dynamicHeight={true} autoPlay>
                    {typeof partners !== undefined && partners.map((item, index) => {
                        return (
                            <div key={index} style={{height: 500}}>
                                <img src={item.image} alt="partner not available"/>
                                <p style={{backgroundColor:"gold"}}className="legend">{item.partner}</p>
                            </div>
                        )
                    })}
                </Carousel>
            </Grid>
            <Grid item xs={false} sm={2}></Grid>
        </Grid>
    )
}

export default Partners