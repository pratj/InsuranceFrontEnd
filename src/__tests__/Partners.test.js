import { shallow } from "enzyme";
import Partners from "../components/Partners";
import { render, screen } from "@testing-library/react";
import toJson from "enzyme-to-json";

test("renders partners component", () => {
  shallow(<Partners />);
});

test("renders partners component with parent div", () => {
  const wrapper = shallow(<Partners />);
  expect(wrapper.find('[data-test="partners"]').length).toBe(1);
});

test("renders Our Partners heading", () => {
  render(<Partners />);
  expect(screen.getByText(/Our Partners/i)).toBeInTheDocument();
});

test("Partners snapshot", () => {
  const partnersTree = shallow(<Partners />);
  expect(toJson(partnersTree)).toMatchSnapshot();
});