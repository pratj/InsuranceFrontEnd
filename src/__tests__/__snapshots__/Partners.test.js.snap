// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Partners snapshot 1`] = `
<WithStyles(ForwardRef(Grid))
  className="partners"
  container={true}
  data-test="partners"
>
  <WithStyles(ForwardRef(Grid))
    item={true}
    sm={2}
    xs={false}
  />
  <WithStyles(ForwardRef(Grid))
    item={true}
    sm={8}
    xs={12}
  >
    <WithStyles(ForwardRef(Typography))
      style={
        Object {
          "color": "#b58500",
          "textAlign": "center",
        }
      }
      variant="h3"
    >
      <b>
        Our Partners
      </b>
    </WithStyles(ForwardRef(Typography))>
    <Carousel
      autoPlay={true}
      axis="horizontal"
      centerSlidePercentage={80}
      dynamicHeight={true}
      interval={3000}
      labels={
        Object {
          "item": "slide item",
          "leftArrow": "previous slide / item",
          "rightArrow": "next slide / item",
        }
      }
      onChange={[Function]}
      onClickItem={[Function]}
      onClickThumb={[Function]}
      onSwipeEnd={[Function]}
      onSwipeMove={[Function]}
      onSwipeStart={[Function]}
      preventMovementUntilSwipeScrollTolerance={false}
      renderArrowNext={[Function]}
      renderArrowPrev={[Function]}
      renderIndicator={[Function]}
      renderItem={[Function]}
      renderThumbs={[Function]}
      selectedItem={0}
      showArrows={true}
      showIndicators={true}
      showStatus={true}
      showThumbs={true}
      statusFormatter={[Function]}
      stopOnHover={true}
      swipeScrollTolerance={5}
      swipeable={true}
      transitionTime={350}
      verticalSwipe="standard"
      width="100%"
    />
  </WithStyles(ForwardRef(Grid))>
  <WithStyles(ForwardRef(Grid))
    item={true}
    sm={2}
    xs={false}
  />
</WithStyles(ForwardRef(Grid))>
`;
